skimage/io/tests/test_tifffile.py

```diff
@@ -27,7 +27,7 @@ def test_imread_uint16():
 def test_imread_uint16_big_endian():
     expected = np.load(fetch('data/chessboard_GRAY_U8.npy'))
     img = imread(fetch('data/chessboard_GRAY_U16B.tif'))
-    assert img.dtype == np.uint16
+    assert img.dtype == np.dtype("<u2")
```

Comment:
This failed previously because `np.dtype(">u2") == np.uint16` (regardless of endian-ness) will always be false. 

I'm not sure what the original intention of this test was? If it was asserting that big endian is returned, then tifffile seems to have been returning the wrong endian-ness for some time. Otherwise, the test should use `img.dtype.type == np.uint16` to check just the dtype without comparing endians.

Reply:
`np.issubdtype`?

Reply:
`img.dtype.type == np.uint16`, like you suggest, feels nice and clean.

Reply:
I take from your last comment then that the test is about asserting `np.uint16` and that we don't care about the endian-ness. I'll update.